memfs/memory.go

```diff
@@ -69,21 +75,40 @@ func (fs *Memory) OpenFile(filename string, flag int, perm os.FileMode) (billy.F
 	return f.Duplicate(filename, perm, flag), nil
 }
 
+func (fs *Memory) resolvePath(path string) string {
+	fullpath := clean(fs.Join(fs.base, path))
+	l, ok := fs.links[fullpath]
+	if !ok {
+		return fullpath
+	}
+
+	// IsAbs requires an unit so we check to if the path stars with a os.PathSeparator
```

Comment:
What is a unit here?

Can you rewrite this, please?

Are you trying to use logical short-circuiting here?